docs/json-codec-backend/codec-docs.md

```diff
@@ -0,0 +1,67 @@
+## Json Codec Backend
+The purpose of this documentation is to give a explanation of how the Json Codec Backend works and how to use it.
+The Json Codec backend is a tool that generates encoders and decoders from a Morphir IR using the Circe JSON library.
+
+### How to use the JSON Codec Backend
+
+The first step to using the JSON Codec backend is to add the Circe JSON library to your project. This can be done in two ways:
+1. Add the Circe JSON dependency through your build tool eg. Sbt, Mill etc.
+2. Add ```morphir-jvm``` as a dependency to your project.
+
+#### Generating Codecs from the IR
+The backend is built to generate codecs for only types so when generating an IR the ```-t``` which specifies ```--types-only``` flag should be used. Also, we want to add the ```-f``` flag to use the old CLI make function.
```

Comment:
This only applies to running `make` on the IR and it is to avoid running into issues with unsupported types in some of the functions. In general, users who just want to generate codecs don't need to do this because the backend will simply only be applied to types and functions will be ignored. I think it would be best to remove the `make` related comments from here and simply say that having a `morphir-ir.json` is a prerequisite. 

As for the Morphir IR related instructions, I think we should have a whole separate section for that at the end of this document since that's mostly for contributors while the general instructions are for users.